### PR TITLE
Add glossary

### PR DESCRIPTION
**Pull request by monopole**
and pointers to it from the main README.

Need a place to define terminology.

This is a conversion and update of the content at go/kinflate-glossary

At the last minute, i pulled workflows out of the glossary, to make it a freestanding MD file, to which we can figures.  Perhaps this stuff goes into the main README, but trying to keep that short.

### CODE REVIEW

File: cmd/kinflate/glossary.md [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+[resource]: #resource
+[resources]: #resource
+[rpm]: https://en.wikipedia.org/wiki/Rpm_(software)
+[target]: #target
+[workflow]: workflows.md
+
+## application
+
+An _application_ is a group of k8s resources that
+server some common purpose, e.g.  a webserver backed by
```
Comment (droot) [inline cmd/kinflate/glossary.md]: s/server/serves

File: cmd/kinflate/glossary.md [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+application resource is just another resource.  There's
+some conceptual overlap between the proposal’s
+application resource and the kinflate [manifest].  The
+application resource has a `Components` field that
+serves a purpose similar to the `resources` field in
+the kinflate manifest.  This overlap can be resolved in
+various ways, the simplest being that kinflate does
+nothing special with the application resource.
+
+
```
Comment (droot) [inline cmd/kinflate/glossary.md]: hmm... not sure if should include "application" here. Do we refer to it anywhere ? Users might think that `kinflate` does something with `application` and the fact it does not.

File: cmd/kinflate/glossary.md (hunk 1 above)
Comment (monopole) [reply cmd/kinflate/glossary.md]: done

File: cmd/kinflate/glossary.md (hunk 2 above)
Comment (monopole) [reply cmd/kinflate/glossary.md]: * Brian's doc is still called declarative application management (not declarative configuration mgmt), and we're an impl, so we kinda have to address the "a" word

* we've been asked to describe how kinflate works with the proposed application resource, both by folks here and folks in the app-def WG.  Trying to answer that by saying that at the moment it does nothing special.

Comment (droot) [reply cmd/kinflate/glossary.md]: Sounds good.

**Pull request merged**